[Swift in WebKit] Fix some missing includes and compile guards in WTF

rr-codes opened the pull request:
#### 543dd66c35df334b4120842ec386780b3e1c4dde
<pre>
[Swift in WebKit] Fix some missing includes and compile guards in WTF
<a href="https://bugs.webkit.org/show_bug.cgi?id=301736">https://bugs.webkit.org/show_bug.cgi?id=301736</a>
<a href="https://rdar.apple.com/163760601">rdar://163760601</a>

Reviewed by Aditya Keerthi.

Fix some issues in WTF that prevent proper modularization:

- Add some missing include statements (since all headers must be self-contained)
- Add some missing Cocoa / ObjC guards (since all headers must work with both C++ and Obj-C++)

* Source/WTF/wtf/AbstractCanMakeCheckedPtr.h:
* Source/WTF/wtf/BlockObjCExceptions.h:
* Source/WTF/wtf/NotFound.h:
* Source/WTF/wtf/ObjCRuntimeExtras.h:
* Source/WTF/wtf/TypeCasts.h:
* Source/WTF/wtf/TypeTraits.h:
* Source/WTF/wtf/ValidatedReinterpretCast.h:
* Source/WTF/wtf/VectorHash.h:
* Source/WTF/wtf/cocoa/NSStringExtras.h:
* Source/WTF/wtf/cocoa/NSURLExtras.h:
* Source/WTF/wtf/cocoa/TypeCastsCocoa.h:
* Source/WTF/wtf/cocoa/VectorCocoa.h:
* Source/WTF/wtf/spi/cocoa/NSLocaleSPI.h:
* Source/WTF/wtf/spi/cocoa/NSObjCRuntimeSPI.h:
* Source/WTF/wtf/spi/darwin/ReasonSPI.h:
* Source/WTF/wtf/spi/mac/MetadataSPI.h:
* Source/WTF/wtf/text/cf/TextBreakIteratorCF.h:
* Source/WTF/wtf/text/cf/TextBreakIteratorCFCharacterCluster.h:
* Source/WTF/wtf/text/cf/TextBreakIteratorCFStringTokenizer.h:
* Source/WTF/wtf/text/cocoa/ContextualizedCFString.h:
* Source/WTF/wtf/text/cocoa/ContextualizedNSString.h:

Canonical link: <a href="https://commits.webkit.org/302374@main">https://commits.webkit.org/302374@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/98f9db7727859dea8937ad872b09d693289f7672

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/128942 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/1195 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/39773 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/136321 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/80299 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/0c534989-dc4d-4377-a66b-27cd0953c7a1) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/1135 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/1074 "Built successfully") | [⏳ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/WPE-WK2-Tests-EWS "Waiting to run tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/66074 "") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/131889 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/869 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/115503 "Passed tests") | [⏳ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/API-Tests-WPE-EWS "Waiting to run tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/74ac8a97-4ec5-4951-ad61-e0c8368c76b2) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/800 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/33617 "Build is in progress. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running run-layout-tests-in-stress-mode; Running layout-tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/79601 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/120934 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/109237 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34113 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/138787 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/127392 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/996 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/971 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/106700 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1057 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/111842 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/106517 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27111 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/827 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/30363 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/53458 "Build is in progress. Recent messages:Printed configuration; Running apply-patch; Checked out pull request; Running compile-webkit") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/1072 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/64392 "") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/160410 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/907 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/40041 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/963 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1003 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WTF/wtf/ValidatedReinterpretCast.h:
```diff
@@ -27,8 +27,8 @@
+#include <wtf/Platform.h>
 #include <bit>
 #include <cstdint>
 #include <wtf/text/ASCIILiteral.h>
```
Please fix the sort order.

rr-codes committed:
[Swift in WebKit] Fix some missing includes and compile guards in WTF
https://bugs.webkit.org/show_bug.cgi?id=301736
rdar://163760601

Reviewed by Aditya Keerthi.

Fix some issues in WTF that prevent proper modularization:

- Add some missing include statements (since all headers must be self-contained)
- Add some missing Cocoa / ObjC guards (since all headers must work with both C++ and Obj-C++)

* Source/WTF/wtf/AbstractCanMakeCheckedPtr.h:
* Source/WTF/wtf/BlockObjCExceptions.h:
* Source/WTF/wtf/NotFound.h:
* Source/WTF/wtf/ObjCRuntimeExtras.h:
* Source/WTF/wtf/TypeCasts.h:
* Source/WTF/wtf/TypeTraits.h:
* Source/WTF/wtf/ValidatedReinterpretCast.h:
* Source/WTF/wtf/VectorHash.h:
* Source/WTF/wtf/cocoa/NSStringExtras.h:
* Source/WTF/wtf/cocoa/NSURLExtras.h:
* Source/WTF/wtf/cocoa/TypeCastsCocoa.h:
* Source/WTF/wtf/cocoa/VectorCocoa.h:
* Source/WTF/wtf/spi/cocoa/NSLocaleSPI.h:
* Source/WTF/wtf/spi/cocoa/NSObjCRuntimeSPI.h:
* Source/WTF/wtf/spi/darwin/ReasonSPI.h:
* Source/WTF/wtf/spi/mac/MetadataSPI.h:
* Source/WTF/wtf/text/cf/TextBreakIteratorCF.h:
* Source/WTF/wtf/text/cf/TextBreakIteratorCFCharacterCluster.h:
* Source/WTF/wtf/text/cf/TextBreakIteratorCFStringTokenizer.h:
* Source/WTF/wtf/text/cocoa/ContextualizedCFString.h:
* Source/WTF/wtf/text/cocoa/ContextualizedNSString.h:

Canonical link: https://commits.webkit.org/302374@main
